src/main/java/org/folio/circulation/resources/ChangeDueDateResource.java

```diff
@@ -0,0 +1,107 @@
+package org.folio.circulation.resources;
+
+import static org.folio.circulation.support.Result.failed;
+import static org.folio.circulation.support.Result.succeeded;
+import static org.folio.circulation.support.ValidationErrorFailure.singleValidationError;
+
+import java.util.concurrent.CompletableFuture;
+
+import org.folio.circulation.domain.Loan;
+import org.folio.circulation.domain.LoanAndRelatedRecords;
+import org.folio.circulation.domain.LoanRepository;
+import org.folio.circulation.domain.notice.schedule.DueDateScheduledNoticeService;
+import org.folio.circulation.domain.representations.ChangeDueDateRequest;
+import org.folio.circulation.domain.validation.ItemStatusValidator;
+import org.folio.circulation.domain.validation.LoanValidator;
+import org.folio.circulation.support.Clients;
+import org.folio.circulation.support.NoContentResult;
+import org.folio.circulation.support.Result;
+import org.folio.circulation.support.RouteRegistration;
+import org.folio.circulation.support.http.server.WebContext;
+import org.joda.time.DateTime;
+
+import io.vertx.core.http.HttpClient;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.web.Router;
+import io.vertx.ext.web.RoutingContext;
+
+public class ChangeDueDateResource extends Resource {
+  public static final String DUE_DATE = "dueDate";
+
+  public ChangeDueDateResource(HttpClient client) {
+    super(client);
+  }
+
+  @Override
+  public void register(Router router) {
+    new RouteRegistration("/circulation/loans/:id/change-due-date", router)
+      .create(this::changeDueDate);
+  }
+
+  private void changeDueDate(RoutingContext routingContext) {
+    createChangeDueDateRequest(routingContext)
+      .after(r -> processChangeDueDate(r, routingContext))
+      .thenApply(NoContentResult::from)
```

Comment:
Hi @marcjohnson-kint, @jeremythuff, are we going to respond with `No content` here or return the updated loan?

Reply:
I don't really mind. I imagine the existing PUT responds with `No content`, which would make this the smaller change.

A challenge with responding with the loan, is that this API is not the loan resource, it is a separate resource. If FOLIO used redirects and locations, then I might suggest using those. Otherwise, if we respond with the loan I think we need to make it clear that it is different to the loan resource itself.

Reply:
Good point. It will add coupling between loan schema and the API. Let's respond with no content then.

Resolving conversation.